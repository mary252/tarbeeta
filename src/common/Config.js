// export const baseURL = "https://api.tarbeeta.com/v1/";
// export const v2_url = "https://api.tarbeeta.com/v2/";
export const v2_url = "https://staging.tarbeeta.com/v2/";
export const baseURL = "https://staging.api.tarbeeta.com/v1/";
export const socketUrl = "https://socket.tarbeeta.com:8878";
// export const baseURL = "http://192.168.1.15:3000/v1/";
// export const baseURL = "http://197.167.99.223:3126/v1/";
 //export const baseURL = "http://msi-laptop:3000/v1/";
export const assetUrl = "https://static.tarbeeta.com/images/";
export const baseUrlStatic = "https://www.tarbeeta.com";

//export const google_capcha="6Lf34qkUAAAAAKBh0Qs_9OZrBeZhikwqF6L7G62p"
// export const google_capcha="6LfE5akUAAAAAHC3KCKqJh1NGup58xJwQbjZxYgg"
export const google_capcha = "6LfXz6oUAAAAABbSRcTAPsK971UiPQZ6CpSSPWr4";
export const EN = 1;
export const AR = 2;
export const EG = 1;
export const DEFAULT_LOCALE = "ar";

// Facebook
export const FACEBOOK_APP_ID = "520130485093535";
export const FACEBOOK_PAGE_ID = "1996928467010956";
export const FACEBOOK_CHAT_POPUP_THEME_COLOR = "#5a9ae5";

export const PRODUCT_SLIDER_LIMIT = 18;

export const ORDER_STATUS_FAILED = 1;
export const ORDER_STATUS_PLACED = 2;
export const ORDER_STATUS_PAID = 3;
export const ORDER_STATUS_PENDING_PAYMENT = 4;
export const ORDER_STATUS_PENDING_SHIPPING = 5;
export const ORDER_STATUS_ON_THE_WAY = 6;
export const ORDER_STATUS_DELIVERED = 7;
export const ORDER_STATUS_CANCELLED = 8;
export const ORDER_STATUS_PENDING = 9;

export const APPROVED_PRODUCT = 4;

export const CASH_ON_DELEVERY = 4;
export const PAY_WALLET = 3;
export const MASTER_CARD = 1;
export const KIOSK = 2;

export const SORT_MOST_RECENT = 1;
export const SORT_OLD_TO_NEW = 2;
export const SORT_HIGHEST_PRICE = 3;
export const SORT_LOWEST_PRICE = 4;
export const SORT_RATING = 5;
export const CARD_PAYMENT_IFRAME = `https://accept.paymobsolutions.com/api/acceptance/iframes/`;
export const EN_FRAME_ID = 7340;
export const AR_FRAME_ID = 7459;
export const pending_product = 1;
