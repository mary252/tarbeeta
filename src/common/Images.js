const Images = {
  aman: require("../assets/images/bitmapAman.png"),
  masary: require("../assets/images/bitmap.png"),
  creditCard: require("../assets/images/cards.png"),
  imagePlaceHolder: require("../assets/images/image-grey-copy-2.png"),
  imagePlaceHolder2: require("../assets/images/image-grey-copy-2@3x.png"),

  slider8Ar: require(`../assets/images/sliders/desktop/ar/8.png`),
  slider8ArMobile: require(`../assets/images/sliders/mobile/ar/8.png`),

  slider8En: require(`../assets/images/sliders/desktop/en/8.png`),
  slider8EnMobile: require(`../assets/images/sliders/mobile/en/8.png`),

  slider9Ar: require(`../assets/images/sliders/desktop/ar/9.png`),
  slider9ArMobile: require(`../assets/images/sliders/mobile/ar/9.png`),

  slider9En: require(`../assets/images/sliders/desktop/en/9.png`),
  slider9EnMobile: require(`../assets/images/sliders/mobile/en/9.png`),

  superMarketAr: require(`../assets/images/sliders/desktop/ar/supermarket-banner-02-ar.jpg`),
  superMarketEn: require(`../assets/images/sliders/desktop/en/supermarket-banner-05-en.jpg`),

  sliderSMMobileAr: require(`../assets/images/sliders/mobile/ar/supermarket-banner-04-ar.jpg`),
  sliderSMMobileEn: require(`../assets/images/sliders/mobile/en/supermarket-banner-06-en.jpg`),

  slider10Ar: require(`../assets/images/sliders/desktop/ar/10.gif`),
  slider10ArMobile: require(`../assets/images/sliders/mobile/ar/10.gif`),

  slider10En: require(`../assets/images/sliders/desktop/en/10.gif`),
  slider10EnMobile: require(`../assets/images/sliders/mobile/en/10.gif`),

  redHeart: require("../assets/images/read-heart.svg"),
  fillHeart: require("../assets/images/heart_filled.svg"),

  notFound404: require("../assets/images/404.svg"),
  noOrders: require("../assets/images/no-orders.svg"),
  emptyCart: require("../assets/images/empty-cart.svg"),
  emptyNotifications: require("../assets/images/empty-notifications.svg"),
  emptySearch: require("../assets/images/empty-search.svg"),
  emptyFavourites: require("../assets/images/no-favourites.svg")
};

export { Images };
