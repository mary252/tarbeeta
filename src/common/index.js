export * from "./Images";
export * from "./Config";
