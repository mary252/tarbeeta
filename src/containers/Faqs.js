import React, { Component } from "react";
import {
  PageTitle,
  Page,
  Faq,
  BreadCrumb,
  Exclamation,
  Payment,
  Setting,
  Rewards,
  Shipping
} from "../components";
import { getFAQs } from "../services";

const RenderChildrens = props => {
  let renderChild = null;
  React.Children.forEach(props.children, child => {
    if (child.key === props.name) {
      renderChild = child;
    }
  });
  return renderChild;
};

const Icon = ({ name }) => (
  <RenderChildrens name={name}>
    <Exclamation key={"Exclamation"} />
    <Payment key={"Payment"} />
    <Setting key={"Setting"} />
    <Rewards key={"Rewards"} />
    <Shipping key={"Shipping"} />
  </RenderChildrens>
);
class Faqs extends Component {
  state = {
    faqs: []
  };

  async componentDidMount() {
    const { lang_id } = this.props;

    await this.getFAQs(lang_id);
  }
  renderFaqs = () =>
    this.state.faqs.map((faq, i) => (
      <Faq
        className="column is-6-desktop is-12-mobile is-12-tablet "
        title={faq.title}
        questions={faq.questions}
        icon={<Icon name={faq.icon} />}
        key={i}
      />
    ));

  getFAQs = async lang_id => {
    try {
      let res = await getFAQs(lang_id);
      this.setState({ faqs: res.data });
    } catch (e) {}
  };
  render() {
    const { locale } = this.props;
    return (
      <Page title={locale.faq} description={locale.faq_desc} {...this.props}>
        <div className="section">
          <div className="container">
            <BreadCrumb
              history={[
                {
                  name: locale.tarbeeta,
                  href: `/${this.props.lang}`
                }
              ]}
              activeRoute={locale.faq}
            />
            <PageTitle title={locale.faq} label={locale.items_label} />
            <div className="columns  is-multiline faq-container">
              {this.renderFaqs()}
            </div>

            <div />
          </div>
        </div>
      </Page>
    );
  }
}

export default Faqs;
