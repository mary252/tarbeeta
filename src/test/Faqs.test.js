import React from "react";
import { shallow } from "enzyme";
import Faqs from "../containers/Faqs";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { baseURL } from "../common/Config";
import axios from "axios"; // v0.15.3
import httpAdapter from "axios/lib/adapters/http";
import nock from "nock";

configure({ adapter: new Adapter() });

const match = {
  params: {
    lang: "en" //any id you want to set
  }
};
const location = {
  search: ""
};

const host = baseURL;

axios.defaults.host = host;
axios.defaults.adapter = httpAdapter;
let query = {
  lang_id: 1
};

describe("Faqs Container", () => {
  const faqs = shallow(<Faqs match={match} location={location} />);
  let instance = faqs.instance();

  it("Should Fetch Faqs And Update the State", () => {
    nock(host)
      .get("/faq")
      .query(query)

      .reply(200, {
        data: [1, 2, 3, 4, 5]
      });
    instance
      .componentDidMount()
      .then(res => expect(instance.state.faqs).toEqual([1, 2, 3, 4, 5]));
  });

  it("  Faqs container exists", () => {
    expect(faqs.exists()).toEqual(true);
  });

  it("renders correctly", () => {
    expect(faqs).toMatchSnapshot();
  });
});
