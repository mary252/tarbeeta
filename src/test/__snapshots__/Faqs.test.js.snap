// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Faqs Container renders correctly 1`] = `ShallowWrapper {}`;
